rust_src/src/eval.rs

```diff
@@ -585,4 +588,91 @@ fn signal_error(msg: &str, arg: LispObject) -> ! {
     );
 }
 
+/// Non-nil if FUNCTION makes provisions for interactive calling.
+/// This means it contains a description for how to read arguments to give it.
+/// The value is nil for an invalid function or a symbol with no function
+/// definition.
+///
+/// Interactively callable functions include strings and vectors (treated
+/// as keyboard macros), lambda-expressions that contain a top-level call
+/// to `interactive', autoload definitions made by `autoload' with non-nil
+/// fourth argument, and some of the built-in functions of Lisp.
+///
+/// Also, a symbol satisfies `commandp' if its function definition does so.
+///
+/// If the optional argument FOR-CALL-INTERACTIVELY is non-nil,
+/// then strings and vectors are not accepted.
+#[lisp_fn(min = "1")]
+pub fn commandp(function: LispObject, for_call_interactively: bool) -> bool {
+    let mut if_prop = false;
+
+    let mut fun = indirect_function(function); // Check cycles.
+    if fun.is_nil() {
+        return false;
+    }
+
+    // Check an `interactive-form' property if present, analogous to the
+    // function-documentation property.
+    while let Some(sym) = fun.as_symbol() {
+        let tmp = get(sym, LispObject::from_raw(Qinteractive_form));
+        if tmp.is_not_nil() {
+            if_prop = true;
+        }
+        fun = symbol_function(sym);
+    }
+
+    if let Some(subr) = fun.as_subr() {
+        // Emacs primitives are interactive if their DEFUN specifies an
+        // interactive spec.
+        if !subr.intspec.is_null() {
+            return true;
+        } else {
+            return if_prop;
+        };
+    } else if let Some(vl) = fun.as_vectorlike() {
+        // Bytecode objects are interactive if they are long enough to
+        // have an element whose index is COMPILED_INTERACTIVE, which is
+        // where the interactive spec is stored.
+        if vl.is_pseudovector(PseudovecType::PVEC_COMPILED)
+            && vl.pseudovector_size() > (COMPILED_INTERACTIVE as i64)
```

Comment:
Is this assuming 64-bit? Should we say `COMPILED_INTERACTIVE as isize` instead?

Reply:
Yes we should use `isize`. The compiler says 'i64' instead of 'EmacsInt' or 'isize'.